### PR TITLE
Don't start minio for Deltalake's ADLS and GCS smoke test

### PR DESCRIPTION
**Pull request by krvikash**
## Description
MinIO is an object storage server compatible with the S3 cloud storage service. It is not specifically designed for ABFS or GCS. To list and delete files in ADLS and GCS, Use the respective file system APIs. It is unnecessary to start minio container for ADLS and GCS smoke tests.

## Release notes

(X) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

**Comment by ebyhr** [comment]
/test-with-secrets sha=10b801dd4f7526c7db5f061585beb47c062a13db

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/HiveMinioDataLake.java [hunk 1]
```diff
@@ -147,6 +157,44 @@ public void close()
         stop();
     }
 
+    private static class HiveHadoopMinioState
```
Comment (ebyhr) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/HiveMinioDataLake.java:160]: How about changing `HiveHadoop hiveMinioDataLake` in `BaseDeltaLakeConnectorSmokeTest` to `HiveHadoop` instead?

Comment (findepi) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/HiveMinioDataLake.java:160]: good point 
we should stop using HiveMinioDataLake if we have to work against convenience this class is supposed to provide

**Comment by krvikash** [comment]
Thanks @ebyhr, @findepi for the review. Addressed comments.

**Comment by ebyhr** [comment]
Could you fix error-prone-checks failure? 
```
 /home/runner/work/trino/trino/plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeAwsConnectorSmokeTest.java:[40,17] [MissingOverride] cleanUp overrides method in BaseDeltaLakeConnectorSmokeTest; expected @Override
```

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeGcsConnectorSmokeTest.java [hunk 2]
```diff
@@ -124,14 +125,15 @@ protected HiveMinioDataLake createHiveMinioDataLake()
         hadoopCoreSiteXmlTempFile.toFile().deleteOnExit();
         Files.writeString(hadoopCoreSiteXmlTempFile, gcpSpecificCoreSiteXmlContent);
 
-        HiveMinioDataLake dataLake = new HiveMinioDataLake(
-                bucketName,
-                ImmutableMap.of(
+        HiveHadoop.Builder hiveHadoopBuilder = HiveHadoop.builder()
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeGcsConnectorSmokeTest.java:128]: nit: We can initialize HiveHadoop without a builder variable. Same for Adls test class:
```java
        HiveHadoop hiveHadoop = HiveHadoop.builder()
                .withImage(HIVE3_IMAGE)
                .withNetwork(closeAfterClass(newNetwork()))
                .withFilesToMount(ImmutableMap.of(
                        "/etc/hadoop/conf/core-site.xml", hadoopCoreSiteXmlTempFile.normalize().toAbsolutePath().toString(),
                        "/etc/hadoop/conf/gcp-credentials.json", gcpCredentialsFile.toAbsolutePath().toString()))
                .build();
```

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeAwsConnectorSmokeTest.java [hunk 3]
```diff
@@ -67,6 +79,15 @@ protected List<String> listCheckpointFiles(String transactionLogDirectory)
                 .collect(toImmutableList());
     }
 
+    @Override
+    protected List<String> listFiles(String directory)
+    {
+        return hiveMinioDataLake.listFiles(directory)
+                .stream()
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeAwsConnectorSmokeTest.java:86]: nit: Move `.stream()` to the previous line. 
```java
        return hiveMinioDataLake.listFiles(directory).stream()
```

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java [hunk 4]
```diff
@@ -144,6 +143,8 @@ protected abstract HiveMinioDataLake createHiveMinioDataLake()
 
     protected abstract List<String> listCheckpointFiles(String transactionLogDirectory);
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java:144]: optional: Overriding `listCheckpointFiles` method looks no longer needed:
```java
    protected List<String> listCheckpointFiles(String transactionLogDirectory)
    {
        return listFiles(transactionLogDirectory).stream()
                .filter(path -> path.contains("checkpoint.parquet"))
                .collect(toImmutableList());
    }
```

**Review by ebyhr — APPROVED** [review]

**Comment by krvikash** [comment]
Thanks, @ebyhr, Addressed comments.

**Pull request merged**